src/sparkx/flow/QCumulantFlow.py

```diff
@@ -0,0 +1,512 @@
+from sparkx.flow import FlowInterface
+import numpy as np
+import random as rd
+import warnings
+
+rd.seed(42)
+
+class QCumulantFlow(FlowInterface.FlowInterface):
+    """
+    **Attention**: This is a beta version of the QCumulantFlow analysis class 
+    for testing purposes. We do not recommend using this class for production 
+    runs yet. There is only the integrated flow implemented at the moment.
+
+    This class implements the Q-cumulant method for anisotropic flow analysis.
+
+    References:
+
+    - [1] `PhD Thesis A. Bilandzic <https://inspirehep.net/literature/1186272>`__
+    - [2] `Phys.Rev.C 83 (2011) 044913 <https://inspirehep.net/literature/871528>`__
+    
+    Parameters
+    ----------
+    n : int, optional
+        The order of the harmonic flow (default is 2).
+    k : int, optional
+        The order of the cumulant (2, 4, or 6) (default is 2).
+    imaginary : str, optional
+        Specifies the treatment of imaginary roots. Options are 'zero', 
+        'negative', or 'nan' (default is 'zero').
+
+    Attributes
+    ----------
+    n_ : int
+        The order of the harmonic flow.
+    k_ : int
+        The order of the cumulant.
+    imaginary_ : str
+        Specifies the treatment of imaginary roots.
+    cumulant_factor_ : dict
+        Dictionary mapping cumulant order to corresponding flow.
+    rand_reaction_planes_ : list
+        List to store randomly sampled reaction planes.
+
+    Methods
+    -------
+    integrated_flow:
+        Computes the integrated flow.
+
+    differential_flow:
+        Computes the differential flow.
+
+    Examples
+    --------
+    A demonstration of how to use the QCumulantFlow class to calculate flow.
+
+    .. highlight:: python
+    .. code-block:: python
+        :linenos:
+
+        >>> flow_instance = QCumulantFlow(n=2, k=2, imaginary='zero')
+        >>> result = flow_instance.integrated_flow(particle_data)
+    """
+    def __init__(self,n=2,k=2,imaginary='zero'):
+        if not isinstance(n, int):
+            raise TypeError('n has to be int')
+        elif n <= 0:
+            raise ValueError('n-th harmonic with value n<=0 can not be computed')
+        else:
+            self.n_ = n
+
+        if not isinstance(k, int):
+            raise TypeError('k has to be int')
+        elif k not in [2,4,6]:
+            raise ValueError(f"{k} particle cumulant is not implemented, choose from [2,4]")
```

Comment:
The list should also contain 6.

Reply:
Oh, I forgot to change this. I started with 2 and 4 and then forgot this part, when I went to 6th order...